src/firebase/api/users.js

```diff
@@ -173,3 +177,113 @@ export const updateUserData = async (updatedData) => {
     };
   }
 };
+
+/**
+ * Sign in user.
+ * @param {String} email email of user.
+ * @param {String} password password of user.
+ * @return {user} authenticated user object.
+ */
+export const signIn = async (email, password) => {
+  try {
+    const userAuth = await auth
+      .signInWithEmailAndPassword(email, password);
+    return {
+      status: StatusCodes.OK,
+      data: userAuth,
+      error: null,
+    };
+  } catch (error) {
+    if (error.code === 'auth/wrong-password' ||
+        error.code === 'auth/user-not-found') {
+      return {
+        status: StatusCodes.NOT_FOUND,
+        data: null,
+        error,
+      };
+    }
+    return {
+      status: StatusCodes.INTERNAL_SERVER_ERROR,
+      data: null,
+      error,
+    };
+  }
+};
+
+/**
+ * Sign current logged in user out.
+ * @return {Object} response object.
+ */
+export const signOut = async () => {
+  try {
+    await auth.signOut();
+    return {
+      status: StatusCodes.OK,
+      data: userAuth,
```

Comment:
`data` should be null here? There is no `userAuth`

Reply:
> a few comments but LGTM! Are these tested btw?

Yes, they are all tested on main scenarios, but I may miss some corner cases, which I will continue testing. I will fix the typo and errors afternoon.